docs/reference/api/connectivity/networksocket/EthInterface.md

```diff
@@ -1,19 +1,45 @@
 ## Ethernet
 
-The EthInterface provides a C++ API for connecting to the internet over Ethernet.
+The `EthInterface` provides a C++ API for connecting to the internet over Ethernet. By default, this class does not require any configuration. It is able to pick up the default Ethernet driver for the target and select the correct network stack.
 
-### EthInterface class reference
+### Usage
 
-[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_eth_interface.html)
+To statically initialize the driver, create an object without passing any parameters:
 
-### Usage
+```cpp
+EthernetInterface eth;
+```
+
+Then, if the default configuration is enough, bring up the interface:
+
+```cpp
+nsapi_error_t status = eth.connect();
+```
+
+Now, the interface is ready to be used for [network sockets](/docs/development/reference/network-socket.html).
+
+```cpp
+// Open a TCP socket
+TCPSocket socket;
+socket.open(&eth);
 
-To bring up the network interface:
+// Open a UDP socket
+UDPSocket socket;
+socket.open(&eth);
+```
 
-1. Instantiate the `EthInterface` class.
-1. Call the `connect` function.
-1. Once you connect the EthInterface, you can use it as a
-target for opening [network sockets](/docs/development/reference/network-socket.html).
+### Configuration
```

Comment:
I think we should move the content from this section into the connectivity configuration section.

Reply:
I don't see that it fits there.
That configuration section is about build time configuration.

This however, refers to the network configuration, how to get the IP address. It is just that you either call the `set_network()` or not... There is nothing to configure through JSON settings.

Should I change the tittle?
Maybe `IP address configuration` would be better to reflect this.